react/components/FacetCheckboxList.js

```diff
@@ -23,62 +26,83 @@ const FacetCheckboxList = ({
   truncatedFacetsFetched,
   setTruncatedFacetsFetched,
 }) => {
-  const { showFacetQuantity } = useContext(SettingsContext)
-  const { getSettings } = useRuntime()
-  const { thresholdForFacetSearch } = useSettings()
-  const [searchTerm, setSearchTerm] = useState('')
-  const [truncated, setTruncated] = useState(true)
-  const isLazyFacetsFetchEnabled = getSettings('vtex.store')
-    ?.enableFiltersFetchOptimization
+  const { push } = usePixel();
+  const { searchQuery } = useSearchPage();
+  const { showFacetQuantity } = useContext(SettingsContext);
+  const { getSettings } = useRuntime();
+  const { thresholdForFacetSearch } = useSettings();
+  const [searchTerm, setSearchTerm] = useState("");
+  const [truncated, setTruncated] = useState(true);
+  const isLazyFacetsFetchEnabled = getSettings("vtex.store")
+    ?.enableFiltersFetchOptimization;
 
   const filteredFacets = useMemo(() => {
-    if (thresholdForFacetSearch === undefined || searchTerm === '') {
-      return facets
+    if (thresholdForFacetSearch === undefined || searchTerm === "") {
+      return facets;
     }
     return facets.filter(
-      facet => facet.name.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1
-    )
-  }, [facets, searchTerm, thresholdForFacetSearch])
+      (facet) => facet.name.toLowerCase().indexOf(searchTerm.toLowerCase()) > -1
+    );
+  }, [facets, searchTerm, thresholdForFacetSearch]);
 
   const shouldTruncate =
-    navigationType === 'collapsible' &&
+    navigationType === "collapsible" &&
     truncateFilters &&
-    // The "+ 1" prevents from truncating a single value 
-    quantity > FACETS_RENDER_THRESHOLD + 1
+    // The "+ 1" prevents from truncating a single value
+    quantity > FACETS_RENDER_THRESHOLD + 1;
 
   const endSlice =
     shouldTruncate && truncated
       ? FACETS_RENDER_THRESHOLD
-      : filteredFacets.length
+      : filteredFacets.length;
 
   const showSearchBar =
     thresholdForFacetSearch !== undefined &&
-    thresholdForFacetSearch < facets.length
+    thresholdForFacetSearch < facets.length;
 
-  const openTruncated = value => {
+  const openTruncated = (value) => {
     if (isLazyFacetsFetchEnabled && !truncatedFacetsFetched) {
-      setTruncatedFacetsFetched(true)
+      setTruncatedFacetsFetched(true);
+    }
+    setTruncated(value);
+  };
+
+  const getCategoryFromObjs = (products) => {
+    const categoryId = products[0].categoryId;
+    const result = products.filter((obj) => obj.categoryId !== categoryId);
+    return result.length == 0 ? categoryId : "";
```

Comment:
Isn't better to return `null` or `undefined` in case it's false?

Reply:
Since it will be printed anyway, in my humble opinion it is better to have an empty striga than undefined or null.